model/dtype.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2017 Makoto Ito
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"math/rand"
+
+	"github.com/chewxy/gorgonia/tensor"
+	"github.com/pkg/errors"
+)
+
+// Dtype stores gorgonia dtype and engine.
+type Dtype struct {
+	T tensor.Dtype
+	E tensor.Engine
+}
```

Comment:
What is good for naming?

Reply:
call it `Type` because you're combining a `Dtype` with its execution engine

Reply:
Sure